hal/src/main/native/include/Simulation/EncoderSim.h

```diff
@@ -123,6 +123,21 @@ class EncoderSim {
     HALSIM_SetEncoderSamplesToAverage(m_index, samplesToAverage);
   }
 
+  std::unique_ptr<CallbackStore> RegisterDistancePerPulseCallback(
+      NotifyCallback callback, bool initialNotify) {
+    auto store = std::make_unique<CallbackStore>(
+        m_index, -1, callback, &HALSIM_CancelEncoderDistancePerPulseCallback);
+    store->SetUid(HALSIM_RegisterEncoderDistancePerPulseCallback(
+        m_index, &CallbackStoreThunk, store.get(), initialNotify));
+    return store;
+  }
+  double GetDistancePerPulse() {
```

Comment:
Please put empty lines between function definitions for readability.

Reply:
Actually, the rest of allwpilib puts newlines between definitions like I requested. If the other simulator classes aren't doing this, they need to be fixed as well (I never reviewed the sim files much, but I help maintain basically everything else and know our prevailing style there). In headers, this grouping can make sense, but these function definitions are longer than one line. Consistency aids readability.

Reply:
#1164 

Doesn't seem like it should be in the scope of this.